ambry-cloud/src/main/java/com/github/ambry/cloud/CloudStorageCompactor.java

```diff
@@ -65,6 +73,29 @@ public void run() {
     }
   }
 
+  /**
+   * Shut down the compactor waiting for in progress operations to complete.
+   */
+  public void shutdown() {
+    shuttingDown.set(true);
+    logger.info("Compactor received shutdown request, waiting up to {} seconds for in-flight operations to finish",
+        shutDownTimeoutSecs);
+    try {
+      doneLatch.get().await(shutDownTimeoutSecs, TimeUnit.SECONDS);
```

Comment:
When a latch times out, it actually doesn't throw an exception, it just returns false, so the return value has to be checked. The InterruptedException is only thrown if the thread is actually interrupted.